lib/logstash/inputs/jms.rb

```diff
@@ -300,6 +336,41 @@ def queue_event(msg, output_queue)
     end
   end
 
+  def to_string_keyed_hash(hash)
+    hash.inject({}) { |h, (key, val)| h[key.to_s] = val; h }
+  end
+  private :to_string_keyed_hash
+
+  # @param msg [JMS::MapMessage]
+  # @return [LogStash::Event]
+  def process_map_message(msg)
+    data = to_string_keyed_hash(msg.data)
+    do_target_check_once_and_get_event_factory.new_event(data)
+  end
+
+  def do_target_check_once_and_get_event_factory
+    @target_checked ||= begin
```

Comment:
note: no synchronization needed because this occurs while iterating over `subscriber.each` from a single thread.